kafkamocks/mock_producer.go

```diff
@@ -0,0 +1,113 @@
+package kafkamocks
+
+import (
+	"errors"
+
+	"github.com/Shopify/sarama"
+)
+
+type TestReporter interface {
+	Errorf(string, ...interface{})
+}
+
+type KafkaProducer interface {
+	AsyncClose()
+	Close() error
+	Input() chan<- *sarama.ProducerMessage
+	Successes() <-chan *sarama.ProducerMessage
+	Errors() <-chan *sarama.ProducerError
+}
+
+var (
+	errProduceSuccess    error = nil
+	errOutOfExpectations       = errors.New("No more expectations set on MockProducer")
+)
+
+type MockProducerExpectation struct {
+	Result error
+}
+
+type MockProducer struct {
+	expectations []*MockProducerExpectation
+	closed       chan struct{}
+	input        chan *sarama.ProducerMessage
+	successes    chan *sarama.ProducerMessage
+	errors       chan *sarama.ProducerError
+}
+
+func NewMockProducer(t TestReporter, config *sarama.Config) *MockProducer {
+	if config == nil {
+		config = sarama.NewConfig()
+	}
```

Comment:
> This reminds me: should we rename Broker in sarama to BrokerConnection or Connection?

... why? I'm not necessarily opposed to `BrokerConnection` on any grounds except it being longer. Just `Connection` seems to lose fairly important context (especially when lazy-connections land) because `Connection.ID()` seems to imply something connection-specific, not broker-specific.